[Binning e2e coverage] Asserting on the saved QB question with explicit joins

nemanjaglumac opened the pull request:
### Status
READY

### What does this PR accomplish?
- Adds initial set of tests around binning, related to starting the new query based on the saved QB (query builder) question with explicit joins

### Screenshots
The current state of tests
![image](https://user-images.githubusercontent.com/31325167/122780263-fadf1b00-d2ae-11eb-995d-f06215c041c6.png)

nemanjaglumac committed:
Add initial binning tests for saved QB question with explicit joins (via simple question)

nemanjaglumac committed:
Add initial binning tests for saved QB question with explicit joins (via custom question)

nemanjaglumac committed:
Add initial binning tests for saved QB question with explicit joins (via column popover)

flamber reviewed (commented):
It looks good, but it's quite difficult to read field IDs instead of constant references. At least all the IDs are in the `table.columns`. But when you define which columns are included via the `query`, then you don't need `table.columns` - example:
```
{"dataset_query":{"type":"query","query":{"source-table":11,"joins":[{"fields":[["field",94,{"join-alias":"People - User"}]],"source-table":12,"condition":["=",["field",80,null],["field",92,{"join-alias":"People - User"}]],"alias":"People - User"}],"fields":[["field",88,null],["field",86,null]]},"database":4},"display":"line","visualization_settings":{}}
```

I would recommend that all the binning is being done on the joined table fields, since I'm guessing that's the point of this test - meaning both time, number and location binning should not be on fields from the base table.

nemanjaglumac commented:
> It looks good, but it's quite difficult to read field IDs instead of constant references.

This is why this PR is still WIP. I'll update hard coded field ids with the variable names, ofc.

> I would recommend that all the binning is being done on the joined table fields, since I'm guessing that's the point of this test - meaning both time, number and location binning should not be on fields from the base table.

Gotcha. I'll update it.
Thank you so much for the feedback. It helps a lot.

nemanjaglumac committed:
Update base question query

nemanjaglumac committed:
Update tests for simple question mode

nemanjaglumac committed:
Update tests for custom question mode

nemanjaglumac committed:
Update tests for the column popover entry mode

nemanjaglumac commented on frontend/test/metabase/scenarios/binning/qb-explicit-joins.cy.spec.js:
```diff
@@ -0,0 +1,328 @@
+        "source-table": ORDERS_ID,
+        joins: [
+          {
+            fields: [
+              ["field", PEOPLE.LONGITUDE, { "join-alias": "People" }],
+              [
+                "field",
+                PEOPLE.BIRTH_DATE,
+                { "temporal-unit": "default", "join-alias": "People" },
+              ],
+            ],
```
Choosing only the specific fields that we need for this test for each of the tables.

nemanjaglumac committed:
Fix comments

nemanjaglumac commented:
@flamber addressed all comments.
1. query now has variable names instead of hard-coded field ids
2. the query itself is simplified and it shows only the needed fields
3. all tests were updated

While writing this, I discovered weird quirk that is documented in #16693.

Weirdly enough, when joining more than once, UI issues from #16675 are not present o.O

flamber reviewed (approved):
I already looked through this, and your simplifications and ID constants looks good.
